account_bank_statement_import_online_qonto/models/online_bank_statement_provider_qonto.py

```diff
@@ -0,0 +1,122 @@
+                date_until,
+            )
+        return self._qonto_obtain_statement_data(date_since, date_until)
+
+    def _qonto_header(self):
+        self.ensure_one()
+        if self.username and self.password:
+            return {'Authorization': '%s:%s' % (self.username, self.password)}
+        raise UserError(_('Please fill login and key'))
+
+    def _qonto_get_slug(self):
+        self.ensure_one()
+        url = QONTO_ENDPOINT + '/organizations/%7Bid%7D'
+        response = requests.get(url, verify=False, headers=self._qonto_header())
+        if response.status_code == 200:
+            data = json.loads(response.text)
+            res = {}
+            for account in data.get('organization', {}).get('bank_accounts', []):
+                iban = sanitize_account_number(account.get('iban', ''))
+                res[iban] = account.get('slug')
+            return res
+        raise UserError(_('%s \n\n %s') % (response.status_code, response.text))
+
+    def _qonto_get_transaction(self, journal, slug, date_since, date_until):
+        self.ensure_one()
+        iban = sanitize_account_number(journal.bank_account_id.acc_number)
+        url = QONTO_ENDPOINT + '/transactions'
+        params = {'slug': slug, 'iban': iban}
+        if date_since:
+            params['settled_at_from'] = date_since.replace(microsecond=0).isoformat() + 'Z'
+        if date_until:
+            params['settled_at_to'] = date_until.replace(microsecond=0).isoformat() + 'Z'
+        response = requests.get(url, verify=False, params=params, headers=self._qonto_header())
+        if response.status_code == 200:
+            return json.loads(response.text)
+        raise UserError(_('%s \n\n %s') % (response.status_code, response.text))
+
+    def _qonto_obtain_statement_data(self, date_since, date_until):
+        self.ensure_one()
+        BankStatement = self.env['account.bank.statement']
+        BankStatementLine = self.env['account.bank.statement.line']
+
+        slugs = self._qonto_get_slug()
+
+        journal = self.journal_id
+
+        slug = slugs.get(sanitize_account_number(journal.bank_account_id.acc_number))
+        if not slug:
+            raise UserError(
+                _('Qonto : wrong configuration, unknow account %s') % journal.bank_account_id.acc_number)
+
+        data = self._qonto_get_transaction(journal, slug, date_since, date_until)
+
+        currency = journal.currency_id or journal.company_id.currency_id
+
+        new_transactions = []
+        sequence = 0
+        for transaction in data.get('transactions', []):
+            if BankStatementLine.sudo().search(
+                [('unique_import_id', '=', transaction['transaction_id']),
+                 ('journal_id', '=', journal.id)], limit=1):
+                continue
+            date = datetime.strptime(transaction['settled_at'], '%Y-%m-%dT%H:%M:%S.%fZ')
+            date = fields.Date.to_date(date.astimezone(pytz.timezone('Europe/Paris')))
```

Comment:
In what timezone the date it returned?

Reply:
always in UTC

Reply:
So why do you convert it to other timezone?

P.S. for better sorting, return datetime in date field, base module will take care of that